src/pages/matrix.tsx

```diff
@@ -1,62 +1,246 @@
+import '@isrd-isi-edu/deriva-webapps/src/assets/scss/_matrix.scss';
+
 import { createRoot } from 'react-dom/client';
+import { useState, useRef, CSSProperties } from 'react';
+import { InputActionMeta } from 'react-select';
 
 // components
 import AppWrapper from '@isrd-isi-edu/chaise/src/components/app-wrapper';
-import Matrix, { MatrixProps } from '@isrd-isi-edu/deriva-webapps/src/components/matrix';
 import ChaiseSpinner from '@isrd-isi-edu/chaise/src/components/spinner';
+import Legend from '@isrd-isi-edu/deriva-webapps/src/components/matrix/legend';
+import VirtualizedGrid from '@isrd-isi-edu/deriva-webapps/src/components/matrix/virtualized-grid';
+import SearchBar from '@isrd-isi-edu/deriva-webapps/src/components/search-bar';
+import VirtualizedSelect from '@isrd-isi-edu/deriva-webapps/src/components/virtualized-select';
 
 // hooks
-import { useEffect, useRef, useState } from 'react';
-import useError from '@isrd-isi-edu/chaise/src/hooks/error';
+import { useWindowSize } from '@isrd-isi-edu/deriva-webapps/hooks/window-size';
+import { useMatrixData } from '@isrd-isi-edu/deriva-webapps/hooks/matrix';
 
 // utilities
 import { ID_NAMES } from '@isrd-isi-edu/chaise/src/utils/constants';
 import { windowRef } from '@isrd-isi-edu/deriva-webapps/src/utils/window-ref';
-import { getConfigObject } from '@isrd-isi-edu/deriva-webapps/src/utils/config';
-
 
 const matrixSettings = {
   appName: 'app/matrix',
   appTitle: 'Matrix',
   overrideHeadTitle: false,
   overrideDownloadClickBehavior: false,
-  overrideExternalLinkBehavior: false
+  overrideExternalLinkBehavior: false,
 };
 
 const MatrixApp = (): JSX.Element => {
-
-  const { dispatchError, errors } = useError();
-
-  const [matrixProps, setMatrixProps] = useState<MatrixProps | null>(null);
-
-  // since we're using strict mode, the useEffect is getting called twice in dev mode
-  // this is to guard against it
-  const setupStarted = useRef<boolean>(false);
-
-  useEffect(() => {
-    if (setupStarted.current) return;
-    setupStarted.current = true;
-
-    try {
-      const config = getConfigObject(windowRef.matrixConfigs);
-      setMatrixProps({ config });
-
-    } catch (error: any) {
-      dispatchError({ error })
-    }
-
-  }, []);
+  const { width = 0, height = 0 } = useWindowSize();
```

Comment:
Let's define these default values in the hook itself.

Reply:
I think this one is ok behavior. if it's undefined it means something else such as no window size exists e.g. executed server side: https://usehooks.com/useWindowSize/